CMakeLists.txt

```diff
@@ -1,3 +1,4 @@
+cmake_minimum_required(VERSION 3.16 FATAL_ERROR)
```

Comment:
Plz confirm that this does not need to be guarded.
Looks good other than that.

Reply:
You can use `cmake_minimum_required(VERSION 3.16...3.22)` if you want to support a range - just make sure you test with 3.22. `FATAL_ERROR` is only needed if you want this to fail on CMake 2.6, which I'm pretty sure has completely vanished (and if someone is using 2.6, they have a lot of other problems!)

It is very much safe and recommended to place this at the top of files, see https://cliutils.gitlab.io/modern-cmake/chapters/basics.html